crates/iota/src/name_commands.rs

```diff
@@ -583,7 +584,11 @@ impl NameCommand {
                 {
                     bail!("target address is already set to the given value");
                 }
-                let nft = get_proxy_nft_by_name(&domain, context).await?;
+                let Some(nft) = get_proxy_nft_by_id_opt(entry.id, &domain, context).await? else {
```

Comment:
You could have just checked if the entry you already have is a leaf record or not

Reply:
Oh, I forgot there is something special about leaf expirations :) I'll change it then, thanks

Reply:
All the new stuff is not really necessary anymore then

Reply:
Not necessary, but better to search by ID than iterate over all owned objects when we can